soc/arm/aspeed/aspeed_util.h

```diff
@@ -0,0 +1,86 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Copyright (c) 2021 ASPEED Technology Inc.
+ */
+#ifndef ZEPHYR_SOC_ARM_ASPEED_UTIL_H_
+#define ZEPHYR_SOC_ARM_ASPEED_UTIL_H_
+#include <sys/util.h>
+#include <devicetree.h>
+#include <toolchain/gcc.h>
+
+/* gcc.h doesn't define __section but checkpatch.pl will complain for this. so
+ * temporarily add a macro here.
+ */
+#ifndef __section
+#define __section(x)    __attribute__((__section__(x)))
+#endif
+
+/* to make checkpatch.pl happy */
+#define ALIGNED16_SECTION(name)         (aligned(16), section(name))
+#define __section_aligned16(name)       __attribute__(ALIGNED16_SECTION(name))
+
+/* non-cached (DMA) memory */
+#if (CONFIG_SRAM_NC_SIZE > 0)
+#define NON_CACHED_BSS                  __section(".nocache.bss")
+#define NON_CACHED_BSS_ALIGN16          __section_aligned16(".nocache.bss")
+#else
+#define NON_CACHED_BSS
+#define NON_CACHED_BSS_ALIGN16          __aligned(16)
+#endif
```

Comment:
You can set `CONFIG_NOCACHE_MEMORY` instead

Reply:
> This SoC needs a board definition to go with it (preferably a second commit added to the PR); otherwise none of this code will get built in CI.

Got it.  I am preparing the second commit and will add it to this PR once I finish it.

>you can set CONFIG_NOCACHE_MEMORY instead

`CONFIG_NOCACHE_MEMORY` depends on `CONFIG_ARCH_HAS_NOCACHE_MEMORY_SUPPORT` which is only enabled on Cortex-M7 for ARM Cortex-M processors.  The CPU in AST10X0 series SOC is Cortex-M3 so I couldn't set  `CONFIG_NOCACHE_MEMORY` for our chip.  Could you kindly advice me if there is a better way to handle this?

Reply:
Force push to add second commit for the ast1030 evaluation board support

Reply:
@MaureenHelm,
Should I resolve the conversation? Or would kindly advice me?